elements/pl-graph/pl-graph.py

```diff
@@ -33,35 +35,30 @@ def graphviz_from_adj_matrix(element, data):
     # Sanity checking
 
     if (mat.shape[0] != mat.shape[1]):
-        raise Exception(f'Non-square adjacency matrix of size ({mat.shape[0]}, {mat.shape[1]}) given as input.')
+        raise ValueError(f'Non-square adjacency matrix of size ({mat.shape[0]}, {mat.shape[1]}) given as input.')
 
     if label is not None:
         mat_label = label
         if (mat_label.shape[0] != mat.shape[0]):
-            raise Exception(f'Dimension of the label ({mat_label.shape[0]}) is not consistent with the dimension of the matrix ({mat.shape[0]})')
+            raise ValueError(f'Dimension of the label ({mat_label.shape[0]}) is not consistent with the dimension of the matrix ({mat.shape[0]})')
     else:
         mat_label = range(mat.shape[1])
 
     # Auto detect showing weights if any of the weights are not 1 or 0
 
     if show_weights is None:
-        all_ones = True
-        for x in mat.flatten():
-            if x != 1 and x != 0:
-                all_ones = False
-        show_weights = not all_ones
+        show_weights = any(
+            x not in {0, 1} for x in mat.flatten()
+        )
 
     # Create pygraphviz graph representation
 
     G = pygraphviz.AGraph(directed=True)
+    G.add_nodes_from(mat_label)
 
-    for node in mat_label:
-        G.add_node(node)
-
-    for i, out_node in enumerate(mat_label):
-        for j, in_node in enumerate(mat_label):
-            x = mat[j, i]
-            if (x > 0):
+    for in_node, row in zip(mat_label, mat):
+        for out_node, x in zip(mat_label, row):
+            if (show_negative_weights) or x > 0:
```

Comment:
```suggestion
            if show_negative_weights or x > 0:
```

Reply:
@stevenxpan I'm realizing that there should be a way to hide edges even when there are negative weights. Do you think that hiding edges when the matrix entries are equal to `nan` is a good idea?